[profiler] Default activities value

ilia-cher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #51632 Fix attribution of some CUDA events to CPU events
* **#51561 [profiler] Default activities value**

Summary:
Using CPU as a default activities value
https://github.com/pytorch/pytorch/issues/51337

Differential Revision: [D26198910](https://our.internmc.facebook.com/intern/diff/D26198910)

Commit:
[profiler] Default activities value

Summary:
Using CPU as a default activities value
https://github.com/pytorch/pytorch/issues/51337

[ghstack-poisoned]

Commit:
Update on "[profiler] Default activities value"


Summary:
Using CPU as a default activities value
https://github.com/pytorch/pytorch/issues/51337

Differential Revision: [D26198910](https://our.internmc.facebook.com/intern/diff/D26198910)

[ghstack-poisoned]

ngimel commented on torch/profiler/profiler.py:
```diff
@@ -140,13 +140,11 @@ def __init__(
         if activities:
             self.activities = activities
         else:
+            self.activities = set([ProfilerActivity.CPU])
```
better set default value for activities, that way it will also automatically show up in docs.

ilia-cher replied:
since we're going to add cuda to the default list (when cuda is enabled) then I'll have to do the check in the code anyway

ngimel replied:
Ok, I'd still set the default value to CPU, but up to you, then here you'd need to check only if cuda is available. 
Btw, what happens when user requests CUDA in activities, but cuda is not available? Is it ignored?

ilia-cher replied:
we output an error in this case

Commit:
Update on "[profiler] Default activities value"


Summary:
Using CPU as a default activities value
https://github.com/pytorch/pytorch/issues/51337

Differential Revision: [D26198910](https://our.internmc.facebook.com/intern/diff/D26198910)

[ghstack-poisoned]

Commit:
Update on "[profiler] Default activities value"


Summary:
Using CPU as a default activities value
https://github.com/pytorch/pytorch/issues/51337

Differential Revision: [D26198910](https://our.internmc.facebook.com/intern/diff/D26198910)

[ghstack-poisoned]

Commit:
Update on "[profiler] Default activities value"


Summary:
Using CPU as a default activities value
https://github.com/pytorch/pytorch/issues/51337

Differential Revision: [D26198910](https://our.internmc.facebook.com/intern/diff/D26198910)

[ghstack-poisoned]
